### ISSUE TITLE
Tab completion 32 bit system fails

### ISSUE THREAD

**Issue by Aldlevine** [JuliaLang/IJulia.jl#564]
On a 32 bit system (specifically Windows, but I assume it impacts others), `Int` is an alias for `Int32`. It's used as such here: [handlers.jl#L48](https://github.com/JuliaLang/IJulia.jl/blob/1b17812b9ee2a544821fe4d00cfd9ac93a2491e5/src/handlers.jl#L48). When using tab-completion I'm receiving this error:

```
KERNEL EXCEPTION
MethodError: no method matching chr2ind(::IJulia.Msg, ::String, ::Int64)
Closest candidates are:
  chr2ind(::IJulia.Msg, ::String, ::Int32) at C:\Users\alevine\.julia\v0.6\IJulia\src\handlers.jl:48

Stacktrace:
 [1] complete_request(::ZMQ.Socket, ::IJulia.Msg) at C:\Users\alevine\.julia\v0.6\IJulia\src\handlers.jl:56
 [2] eventloop(::ZMQ.Socket) at C:\Users\alevine\.julia\v0.6\IJulia\src\eventloop.jl:8
 [3] (::IJulia.##11#14)() at .\task.jl:335
```

It looks like, while the method only accepts an `Int32` (in this scenario), the caller passes in an `Int64`. I believe this is from `msg.content["cursor_pos"]` in `complete_request`.

I've been able to work around this by adding the below line at the top of an IJulia notebook. This appears to produce the correct functionality.

```julia
Base.chr2ind(m::IJulia.Msg, str::String, ic::Int64) = Base.chr2ind(m, str, Int32(ic))
```

**Comment by stevengj** [comment]
The right fix here is probably to just change the `chr2ind` signature to `Base.chr2ind(m::Msg, str::String, ic::Integer)` and similarly fro `ind2chr`.

**Comment by xiaoweiz** [comment]
I met a similar issue using Tab Completion in JupyterLab on OSX (jupyterlab/jupyterlab#2768). Not sure if the problem is related.

**Comment by stevengj** [comment]
Should be fixed on master.

**Comment by xiaoweiz** [comment]
I presume the fix doesn't aim for  (jupyterlab/jupyterlab#2768)? I'm not even sure this is an issue on `IJulia` end. Anyway, the error message when using tab completion in JupypterLab (not Jupyter Notebook) is 
```julia
KERNEL EXCEPTION
BoundsError: attempt to access "using"
  at index [0]

Stacktrace:
 [1] next at ./strings/string.jl:197 [inlined]
 [2] getindex(::String, ::Int64) at ./strings/basic.jl:32
 [3] ind2chr(::String, ::Int64) at ./strings/basic.jl:269
 [4] ind2chr(::IJulia.Msg, ::String, ::Int64) at /Users/xiaoweiz/.julia/v0.6/IJulia/src/handlers.jl:50
 [5] complete_request(::ZMQ.Socket, ::IJulia.Msg) at /Users/xiaoweiz/.julia/v0.6/IJulia/src/handlers.jl:76
 [6] eventloop(::ZMQ.Socket) at /Users/xiaoweiz/.julia/v0.6/IJulia/src/eventloop.jl:8
 [7] (::IJulia.##14#17)() at ./task.jl:335
```

**Comment by stevengj** [comment]
The fix in c6a33c40eb0cf463644648988581dc6e8aed95e5 was only for @Aldlevine's issue; the problem on JupyterLab seems unrelated.
